File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/nudg++/trunk/Src/Codes2D/PhysDmatrices2D.cpp
// PhysDmatrices2D.m
// function [Dx,Dy] = PhysDmatrices2D(x1, y1, interp)
// 2007/10/07
//---------------------------------------------------------
#include "NDGLib_headers.h"
#include "NDG2D.h"


//---------------------------------------------------------
void NDG2D::PhysDmatrices2D
(
  const DVec& x1,     // [in]
  const DVec& y1,     // [in]
  const DMat& interp, // [in]
        DMat& Dx,     // [out]
        DMat& Dy      // [out]
)
//---------------------------------------------------------
{
  // function [Dr,Ds] = PhysDmatrices2D(x1, y1, interp)
  // Purpose : Initialize the (x,y) differentiation matrices
  //	    on the simplex, evaluated at (x1,y1) at order N

  DMat_Diag RX, SX, RY, SY;
  DVec      rx1,sx1,ry1,sy1, J1;

  DMat IDr=interp*Dr, IDs=interp*Ds;

  ::GeometricFactors2D(x1, y1, IDr, IDs,      // [in]
                       rx1,sx1,ry1,sy1,J1);   // [out]

  // load vectors as diagonal matrices
  RX = rx1;  SX = sx1;
  RY = ry1;  SY = sy1;

  Dx = RX*IDr + SX*IDs;
  Dy = RY*IDr + SY*IDs;
}

File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/Codes1.1/Codes2D/PhysDmatrices2D.m
function [Dx,Dy] = PhysDmatrices2D(x1, y1, interp)

% function [Dr,Ds] = PhysDmatrices2D(x1, y1, interp)
% Purpose : Initialize the (x,y) differentiation matrices on the simplex
%	        evaluated at (x1,y1) at order N

Globals2D;
IDr = interp*Dr; IDs = interp*Ds;

[rx1,sx1,ry1,sy1,J1] = GeometricFactors2D(x1, y1, IDr, IDs);

n = size(interp, 1);
Dx = spdiags(rx1, 0, n, n)*IDr + spdiags(sx1, 0, n, n)*IDs;
Dy = spdiags(ry1, 0, n, n)*IDr + spdiags(sy1, 0, n, n)*IDs;
return;
